src/module.c

```diff
@@ -9954,9 +9960,10 @@ int RM_DictReplace(RedisModuleDict *d, RedisModuleString *key, void *ptr) {
  * be set by reference to 1 if the key does not exist, or to 0 if the key
  * exists. */
 void *RM_DictGetC(RedisModuleDict *d, void *key, size_t keylen, int *nokey) {
-    void *res = raxFind(d->rax,key,keylen);
-    if (nokey) *nokey = (res == raxNotFound);
-    return (res == raxNotFound) ? NULL : res;
+    void *res;
+    int found = raxFind(d->rax,key,keylen,&res);
+    if (nokey) *nokey = !found;
+    return found ? res : NULL;
```

Comment:
Just using the pattern of raxFind() will not overwrite NULL values to simplify.
```suggestion
    void *res = NULL;
    int found = raxFind(d->rax,key,keylen,&res);
    if (nokey) *nokey = !found;
    return res;
```